bin/webhook-create-signed-cert.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+set -e
+
+usage() {
+    cat <<EOF
+Generate certificate suitable for use with an sidecar-injector webhook service.
+
+This script uses k8s' CertificateSigningRequest API to a generate a
+certificate signed by k8s CA suitable for use with sidecar-injector webhook
+services. This requires permissions to create and approve CSR. See
+https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster for
+detailed explantion and additional instructions.
+
+The server key/cert k8s CA cert are stored in a k8s secret.
```

Comment:
I don't think this will work as a long-term solution:

1. The private key for the certificate shouldn't ever be outside the cluster, so this script should be run from within the controller, not by the user.

2. The CertificateSigningRequest API isn't always enabled and even when enabled isn't always configured in a usable state. For example and in particular, I don't think minikube enables this in its default configuration.

3. AFAICT, using the CertificateSigningRequest API isn't necessary because the webhook configuration YAML lets you specify the CA to be trusted for the webhook.

The way I was thinking to do this is as follows:
1. `conduit install` installs the webhook along with the rest of the control plane but it won't work yet because the object properties that indicates the webhook's CA won't be written yet.
2. When the CA starts up, (and in the future, whenever the root CA certificates change) the CA replaces the webhooks configuration with one that has its own certificate as the trusted CA certificate. This requires the CA to have `edit` properties for the webhook configuration object.

After step 2 the webhook should work.

Reply:
Yeah, this is really the first implementation, if you look at Istio the setup and creation of the certificate is made at the setup time using k8s api.

The point 1 or 2 are prerequisite.

Only one point, i have made all test against clean minikube environment and all is working , so i think the `CertificateSigningRequest` it's enabled by default.

Reply:
> ... creation of the certificate is made at the setup time using k8s api.

Not 100% sure, from what I see in the citadel code, they may no longer be relying on the k8s api to generate and sign CSR. Can you confirm?

References:
1. [security/cmd/istio_ca/main.go#L427-L454)](https://github.com/istio/istio/blob/2645f480a1ee8f8dcf2c29b39894f434b76e9783/security/cmd/istio_ca/main.go#L427-L454)
1. [security/pkg/pki/ca/ca.go#L174-L194](https://github.com/istio/istio/blob/2645f480a1ee8f8dcf2c29b39894f434b76e9783/security/pkg/pki/ca/ca.go#L174-L194)
1. [security/pkg/pki/util/generate_cert.go#L107-L114](https://github.com/istio/istio/blob/2645f480a1ee8f8dcf2c29b39894f434b76e9783/security/pkg/pki/util/generate_cert.go#L107-L114)